### PR TITLE
Kernelwindow: Add optional kernel flavour selector

### PR DESCRIPTION
**Pull request by gm10**
Adds this:
kernel-flavour

Only shows when `com.linuxmint.updates.allow-kernel-type-selection` is set in gsettings, which replaces the old `use-lowlatency-kernels` that was never properly implemented. Whether a GUI toggle in the Preferences window is desired can be decided later.

Includes #434 and https://github.com/linuxmint/mintupdate/commit/f1c76074624dd24327b8e05ae55d6e73957ee026 from #454

### CODE REVIEW

File: usr/lib/linuxmint/mintUpdate/mintUpdate.py [hunk 1]
```diff
@@ -907,19 +907,19 @@ def run(self):
                                 infobar_title = _("Please switch to another mirror")
                                 infobar_message = _("%s is unreachable.") % mirror_url
                         elif mint_timestamp is not None:
-                            mint_date = datetime.datetime.fromtimestamp(mint_timestamp)
-                            now = datetime.datetime.now()
+                            mint_date = datetime.fromtimestamp(mint_timestamp)
+                            now = datetime.now()
```
Comment (clefebvre) [inline usr/lib/linuxmint/mintUpdate/mintUpdate.py]: This is not related to this PR, it shouldn't be here. Also it's incompatible with master.

File: usr/lib/linuxmint/mintUpdate/mintUpdate.py [hunk 2]
```diff
@@ -907,19 +907,19 @@ def run(self):
...
                             mint_age = (now - mint_date).days
                             if (mint_age > 2):
-                                mirror_date = datetime.datetime.fromtimestamp(mirror_timestamp)
+                                mirror_date = datetime.fromtimestamp(mirror_timestamp)
```
Comment (clefebvre) [inline usr/lib/linuxmint/mintUpdate/mintUpdate.py]: Same as above.

File: usr/lib/linuxmint/mintUpdate/mintUpdate.py [hunk 3]
```diff
@@ -955,7 +955,7 @@ def run(self):
                                                                 "The last update on %(mirror)s was %(days)d days ago.",
                                                                 (now - mirror_date).days) % \
                                                                 {'mirror': mirror_url, 'days': (now - mirror_date).days}
-            except Exception as e:
+            except:
```
Comment (clefebvre) [inline usr/lib/linuxmint/mintUpdate/mintUpdate.py]: You're undoing what you did before?

File: usr/lib/linuxmint/mintUpdate/kernelwindow.py [hunk 4]
```diff
@@ -430,19 +478,12 @@ def __init__(self, application):
...
 
-        # Center on main window
-        window_size = self.window.get_size()
-        parent_size = self.application.window.get_size()
-        parent_position = self.application.window.get_position()
-        parent_center_x = parent_position.root_x + parent_size.width / 2
-        parent_center_y = parent_position.root_y + parent_size.height / 2
-        self.window.move(parent_center_x - window_size.width / 2,
-                         parent_center_y - window_size.height / 2)
-
```
Comment (clefebvre) [inline usr/lib/linuxmint/mintUpdate/kernelwindow.py]: Wasn't that added in a previous commit?

File: usr/lib/linuxmint/mintUpdate/mintUpdate.py (hunk 3 above)
Comment (gm10) [reply usr/lib/linuxmint/mintUpdate/mintUpdate.py]: I'll put it into a separate PR

File: usr/lib/linuxmint/mintUpdate/kernelwindow.py (hunk 4 above)
Comment (gm10) [reply usr/lib/linuxmint/mintUpdate/kernelwindow.py]: It's been moved, not removed. My local branch had it in a different spot, so I just moved it there without losing sleep over why positions had changed. My local branch is authoritative for me.
